### PR TITLE
Improvements and fixes for v1.1

### PR DESCRIPTION
**Pull request by joegilkes**
Adds many improvements and necessary fixes:

* `ModelPredictor` now returns its predictions when called within Python
* More scikit-learn MLPRegressor options can be passed through from the command line
* Fixed a bug allowing training data to leak into the test set
* Enforced use of hydrogen information when parsing SMILES
* Added more options for constructing fingerprints, including allowing difference vs concatenation fingerprints, and adding optional features based on the number of atoms in a reaction
* Fixed training data splitting for ensembles
* Added support for early stopping to prevent overfitting

### CODE REVIEW

**Comment by joegilkes** [comment]
Ignore this, was trying to merge into my own fork and didn't realise GitHub had selected yours.
